core/src/avm2/globals/string.rs

```diff
@@ -27,13 +40,98 @@ pub fn class_init<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements `length` property's getter
+fn length<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    _args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let Some(this) = this {
+        if let Value::String(s) = this.value_of(activation.context.gc_context)? {
+            return Ok(s.encode_utf16().count().into());
+        }
+    }
+
+    Ok(Value::Undefined)
+}
+
+/// Implements `String.charAt`
+fn char_at<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    match char_code_at(activation, this, args)? {
+        Value::Number(code) if code.is_finite() => Ok(AvmString::new(
+            activation.context.gc_context,
+            String::from_utf16_lossy(&[code as u16]),
```

Comment:
Let's use our `string_utils::utf16_code_unit_to_char(c)` here since that's what's also used in avm1. I think it's also clearer to duplicate the logic from `char_code_at` here instead of going from char code->Value->char.